### DISCUSSION TITLE
COmparing CAN ID with values

### DISCUSSION THREAD

**Discussion by Admsher** [stm32-rs/stm32f4xx-hal#818] [Q&A] [answered]
I am trying to compare the id for can data coming in with my   u16 values from the data received with this

`let received = block!(can.receive()).unwrap();`

I then fetch the id using this
`received .id()`'
this gives me a datatype with this format :
Extended(ExtendedId(id no.))

is there a way to extract this id no. ??

**Comment by burrbull** [comment]
Please, open discussions for such questions. Or even better connect to matrix chat https://github.com/rust-embedded/awesome-embedded-rust?tab=readme-ov-file#community-chat-rooms
`issues` are for ideas and reporting bugs.

https://docs.rs/bxcan/0.8.0/bxcan/struct.ExtendedId.html#method.as_raw

**Reply by burrbull** [reply, marked as answer]
```rust
let equal = match id {
    Standard(s) => s.as_raw() == my_val,
    Extended(e) => e.as_raw() == my_val as u32, // note that extended Id is 32-bit
};
```
